contrib/docker-integration/run.sh

```diff
@@ -60,21 +56,20 @@ until "$DOCKER_BINARY" -H "127.0.0.1:$ENGINE_PORT" version &> /dev/null; do
 	sleep 1
 done
 
-# Make sure we have images outside the container, to transfer to the container.
-# Not much will happen here if the images are already present.
-docker-compose pull
-docker-compose build
-
-# Transfer images to the inner container.
-for image in "$INTEGRATION_IMAGE" registry:0.9.1 dockerintegration_nginx dockerintegration_registryv2; do
-	docker save "$image" | "$DOCKER_BINARY" -H "127.0.0.1:$ENGINE_PORT" load
-done
-
-#DOCKER_HOST="tcp://127.0.0.1:$ENGINE_PORT" docker-compose pull
-#DOCKER_HOST="tcp://127.0.0.1:$ENGINE_PORT" docker-compose build
+# If no volume is specified, transfer images into the container from
+# the outer docker instance
+if [ "$DOCKER_VOLUME" != "" ]; then
+	# Make sure we have images outside the container, to transfer to the container.
+	# Not much will happen here if the images are already present.
+	docker-compose pull
+	docker-compose build
+
+	# Transfer images to the inner container.
+	for image in "$INTEGRATION_IMAGE" registry:0.9.1 dockerintegration_nginx dockerintegration_registryv2; do
+		docker save "$image" | docker exec -i "$ID" docker load
```

Comment:
Interesting approach - I hadn't thought of running the docker client inside the container. I wonder if this slows down the `docker load` at all, because of the need to pipe data through `docker exec`. It was already quite slow before, and I'm not sure why.